pkg/snapstore/swift_snapstore_test.go

```diff
@@ -30,6 +32,12 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+type bulkDeleteJSONResponse struct {
+	Deleted      int      `json:"deleted"`
+	NotFound     int      `json:"not_found"`
+	ErrorStrings []string `json:"errors,omitempty"`
+}
+
```

Comment:
Is this something you have come with by yourself? The standard [Result struct](https://github.com/gardener/etcd-backup-restore/blob/4f1f191f886124d9c45e8606255c0e00f7b220d9/vendor/github.com/gophercloud/gophercloud/results.go#L14-L39) used by openstack should be more than sufficient, right?

It gives you a `Body`, `Header` and `Err`. `Err` can be formed using errors.Join(), which takes an array/slice of error objects. And if we really need to record notFound and Deleted counts, we can directly write it to the response Body. Unless this struct is absolutely necessary, please remove it, to avoid reinventing the wheel.

Reply:
Oops, I didn't check https://docs.openstack.org/swift/latest/api/bulk-delete.html, apologies for that. I have seen that now, and wonder whether we need to correctly handle the errors from the response `Body` [here](https://github.com/gardener/etcd-backup-restore/blob/1a0dcf14220c452d86d2a8c6b65c26946d7475b3/pkg/snapstore/swift_snapstore.go#L521-L523), rather than response `Err`.

Reply:
The `Result` struct would not be sufficient since we need to extract information about the operation. Instead we should be using the following struct to obtain the relevant information:

https://github.com/gardener/etcd-backup-restore/blob/4f1f191f886124d9c45e8606255c0e00f7b220d9/vendor/github.com/gophercloud/gophercloud/openstack/objectstorage/v1/objects/results.go#L458-L464

Once we `BulkDeleteResult.Extract()` to obtain the `BulkDeleteResponse`, we can check for errors that could have possibly occurred in the `Errors` field.

The error handling as linked by you in `swift_snapstore.go` is not sufficient since the error while performing the `POST` operation is checked for, and returned. None of the errors that could potentially occur server side are checked for.

I will remove the `bulkDeleteJSONResponse` struct and make use of `BulkDeleteResponse` to populate the response's `Body`.